### PR TITLE
Dev/oo#80 merge pdfs together instead of zip

### PR DESCRIPTION
**Pull request by mcmpp**

### CODE REVIEW

File: src/main/scala/ch/openolitor/core/RouteService.scala [hunk 1]
```diff
@@ -445,13 +447,13 @@ trait DefaultRouteService extends HttpService with ActorReferences with BaseJson
...
 
-  protected def downloadAll(zipFileName: String, fileType: FileType, ids: Seq[FileStoreFileId]) = {
+  protected def downloadAll(pdfFileName: String, fileType: FileType, ids: Seq[FileStoreFileId]) = {
     ids match {
       case Seq()            => complete(StatusCodes.BadRequest)
       case Seq(fileStoreId) => download(fileType, fileStoreId.id)
       case list =>
         val refs = ids.map(id => FileStoreFileReference(fileType, id))
-        downloadAsZip(zipFileName, refs)
+        downloadMergedPDFs(pdfFileName, refs)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/core/RouteService.scala]: If we just download the files, there is no possibility to choose PDF/ZIP?

File: src/main/scala/ch/openolitor/core/filestore/FileStore.scala [hunk 2]
```diff
@@ -165,6 +165,7 @@ trait FileStore {
 class S3FileStore(override val mandant: String, mandantConfiguration: MandantConfiguration, actorSystem: ActorSystem) extends FileStore with FileStoreBucketLifeCycleConfiguration with LazyLogging {
   val opts = new ClientConfiguration
   opts.setSignerOverride("S3SignerType")
+  opts.setMaxConnections(1000)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/core/filestore/FileStore.scala:171]: One thought about this problem: Could it be, that we have a parallel map execution? Is the order of the pages in resulting PDF always the same?

File: src/main/scala/ch/openolitor/core/RouteService.scala (hunk 1 above)
Comment (mcmpp) [reply src/main/scala/ch/openolitor/core/RouteService.scala]: no, on the "download" option there is no menu. Do you think it makes sense to add one??

File: src/main/scala/ch/openolitor/core/filestore/FileStore.scala (hunk 2 above)
Comment (mcmpp) [reply src/main/scala/ch/openolitor/core/filestore/FileStore.scala:168]: it is parallel. It is not always respecting the same order. Do you think this is a requirement??

Comment (mcmpp) [reply src/main/scala/ch/openolitor/core/filestore/FileStore.scala:171]: I just pushed a solution to get the merged pdf ordered

File: src/main/scala/ch/openolitor/core/RouteService.scala (hunk 1 above)
Comment (AlwinEgger) [reply src/main/scala/ch/openolitor/core/RouteService.scala]: Don't know. I guess it could as handy as in the moment of generating, no?

Comment (mcmpp) [reply src/main/scala/ch/openolitor/core/RouteService.scala]: done

File: src/main/scala/ch/openolitor/core/reporting/ReportService.scala [hunk 3]
```diff
@@ -117,21 +117,22 @@ trait ReportService extends LazyLogging with AsyncConnectionPoolContextAware wit
...
   def generateReport[I, E](vorlage: Array[Byte], data: ReportData[E], pdfGenerieren: Boolean,
-    pdfAblage: Option[FileStoreParameters[E]], downloadFile: Boolean, jobId: JobId)(implicit personId: PersonId): ServiceResult[JobId] = EitherT {
-    logger.debug(s"generateReport: vorlage: $vorlage, pdfGenerieren: $pdfGenerieren, pdfAblage: $pdfAblage, downloadFile: $downloadFile, jobId: $jobId")
+    pdfAblage: Option[FileStoreParameters[E]], downloadFile: Boolean, pdfMerge: Boolean, jobId: JobId)(implicit personId: PersonId): ServiceResult[JobId] = EitherT {
+    logger.debug(s"generateReport: vorlage: $vorlage, pdfGenerieren: $pdfGenerieren, pdfAblage: $pdfAblage, downloadFile: $downloadFile, pdfMerge: $pdfMerge, jobId: $jobId")
     val collector =
-      if (pdfAblage.isDefined) FileStoreReportResultCollector.props(reportSystem, jobQueueService, downloadFile)
+      if (pdfAblage.isDefined) FileStoreReportResultCollector.props(reportSystem, jobQueueService, downloadFile, pdfMerge)
       else if (data.rows.size == 1) HeadReportResultCollector.props(reportSystem, jobQueueService)
+      else if (data.rows.size == 1) HeadReportResultCollector.props(reportSystem, jobQueueService)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/core/reporting/ReportService.scala]: duplicated row

Comment (mcmpp) [reply src/main/scala/ch/openolitor/core/reporting/ReportService.scala]: ups

**Review by AlwinEgger — APPROVED** [review]

**Pull request merged**